src/components/MessagesList/MessagesList.vue

```diff
@@ -1266,6 +1266,10 @@ export default {
 			// setTimeout is needed here for Safari to correctly remove the unread marker
 			setTimeout(() => {
 				this.refreshReadMarkerPosition()
+				// Regenerate relative date separators
+				Object.keys(this.dateSeparatorLabels).forEach(dateTimestamp => {
+					this.$set(this.dateSeparatorLabels, dateTimestamp, this.generateDateSeparator(dateTimestamp))
```

Comment:
you need to call `this.$set` on every labels change, otherwise it's not reactive from the start, when you create it in `prepareMessagesGroups`